sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,407 @@
+this solution.
+
+Step 10 : If `m` is a non-negative integer -
+
+Step 11: Find a polynomial solution of degree `m` for the auxiliary equation.
+
+There are 2 cases possible -
+
+    a. `m` is a non-negative integer: We can solve for the coefficients
+    in `p(x)` using Undetermined Coefficients.
+
+    b. `m` is an expression: In this case, we may not be able to determine
+    if it is possible for `m` to be a non-negative integer. However, the
+    solution to the auxiliary equation always seems to be x**m + C1 (where
+    C1 is an arbitrary constant). This has been confirmed by testing many
+    examples. NOTE that this is NOT mentioned in the thesis and is an
+    assumption made by me. If any failing test cases are found, this case
+    must be removed.
+
+Step 12 : For each `p(x)` that exists, append `ybar + \frac{p'(x)}{p(x)}`
+to ``presol``.
+
+Step 13 : For each solution in ``presol``, apply an inverse transformation
+and append them to ``sol``, so that the solutions of the original equation
+are found using the solutions of the equation in its normal form.
+"""
+
+from sympy.core import S
+from sympy.core.function import Function
+from sympy.core.numbers import oo
+from sympy.core.relational import Eq
+from sympy.core.symbol import Symbol
+from sympy.functions import sqrt
+from sympy.polys.polytools import Poly
+from sympy.polys.polyroots import roots
+from sympy.simplify.radsimp import numer
+from sympy.series.limits import limit
+from sympy.solvers.solvers import solve_undetermined_coeffs
+
+def find_poles(num, den, x):
+    # All roots of denominator are poles of a(x)
+    p = roots(den, x)
+    # Substitute 1/x for x and check if oo is a pole
+    if den.degree(x) > num.degree(x):
+        if num.expr != 0:
+            num = num.transform(Poly(1, x), Poly(x, x)) * x**(den.degree(x) - num.degree(x))
+            den = den.transform(Poly(1, x), Poly(x, x))
+    else:
+        num = num.transform(Poly(1, x), Poly(x, x))
+        den = den.transform(Poly(1, x), Poly(x, x)) * x**(num.degree(x) - den.degree(x))
+    num, den = num.cancel(den)[1:]
+
+    # Find the poles of a(1/x)
+    p_inv = roots(den, x)
+
+    # If 0 is a pole of a(1/x), oo is a pole of a(x)
+    if 0 in p_inv:
+        p.update({oo: p_inv[0]})
+    return p
+
+
+def val_at_inf(num, den, x):
+    # Valuation of a rational function at oo = deg(denom) - deg(numer)
+    return den.degree(x) - num.degree(x)
+
+
+def construct_c(a, x, poles, muls):
+    c = []
+    for pole, mul in zip(poles, muls):
+        c.append([])
+
+        # Case 3
+        if mul == 1:
+            # If multiplicity is 1, the coefficient to be added
+            # in the c-vector is 1 (no choice)
+            c[-1].extend([[1], [1]])
+
+        # Case 1
+        elif mul == 2:
+            # Find the coefficient of 1/(x - pole)**2 in the
+            # Laurent series expansion of a(x) about pole.
+            r = a.series(x, pole).coeff(x - pole, -2)
```

Comment:
We don't need to compute the whole series just for one coefficient:
```
In [9]: f = (x**2 - 1)/(x**2 + 1)/(x - 2)**2

In [10]: f
Out[10]: 
       2         
      x  - 1     
─────────────────
       2 ⎛ 2    ⎞
(x - 2) ⋅⎝x  + 1⎠

In [11]: f.series(x, 2)
Out[11]: 
                                          2             3             4              5                            
    3            8        206   82⋅(x - 2)    88⋅(x - 2)    58⋅(x - 2)    672⋅(x - 2)    48⋅x    ⎛       6       ⎞
────────── + ────────── - ─── - ─────────── + ─────────── + ─────────── - ──────────── + ──── + O⎝(x - 2) ; x → 2⎠
         2   25⋅(x - 2)   625       3125         15625         78125         390625      625                      
5⋅(x - 2)                                                                                                         

In [12]: cancel(f*(x - 2)**2).subs(x, 2)
Out[12]: 3/5
```

Reply:
Got it, didn't realize that